### PR TITLE
Remove HtlcLocation from Ledger trait

### PR DESCRIPTION
**Pull request by thomaseizinger**
Remove the HtlcLocation associated type from Ledger trait.

In preparation for removing the `Ledger` trait remove the `HtlcLocation` trait bound from `Ledger` trait.  This PR is split into 3 patches.

- Patch 1 does the bulk of the work (also implements idea presented in [comment below](https://github.com/comit-network/comit-rs/pull/2128#pullrequestreview-367521434))
- Patch 2 adds a type alias that is WIP i.e., will be removed once we remove `Transaction` from `Ledger`.  Kept separate because its a bit ugly.
- Patch 3 Removes the actual associated type i.e., this patch is a very small diff that explains why we do everything in patch 1.

Please see commit messages on each patch for more details, thanks.

Works towards: https://github.com/comit-network/comit-rs/issues/2081

### CODE REVIEW

File: cnd/src/db/with_swap_types.rs [hunk 1]
```diff
@@ -55,6 +55,10 @@ macro_rules! with_swap_types {
                     #[allow(dead_code)]
                     type BI = identity::Ethereum;
                     #[allow(dead_code)]
+                    type AH = ::bitcoin::OutPoint;
```
[the remark this answers is not in the document]
Comment (tcharding) [reply cnd/src/db/with_swap_types.rs]: How about `htlc_location::Bitcoin`, `htlc_location::Ethereum`?
```
pub mod htlc_location {
    pub use crate::{bitcoin::OutPoint as Bitcoin, ethereum::Address as Ethereum};
}
```

**Comment by tcharding** [comment]
Woops, I hit `approved` by habit.  Self-requested a review to try to cancel it.

**Comment by thomaseizinger** [comment]
You can always just dismiss the review at the very bottom :)

**Comment by thomaseizinger** [comment]
Re order: Happy to do that once these things are less in flux :)

File: cnd/src/db/with_swap_types.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tcharding) [reply cnd/src/db/with_swap_types.rs]: > Re order: Happy to do that once these things are less in flux :)

The re-order is only concerned with this PR, what do we hope to gain by waiting?

[the remark this answers is not in the document]
Comment (tcharding) [reply cnd/src/db/with_swap_types.rs]: Oh, I'm wrong.  There are various combinations of H, A, T that are in places not touched by this PR.  Will not touch those as suggested.

**Comment by thomaseizinger** [comment]
@tcharding I rebased this onto latest dev, now that all the PRs this built upon are merged.

File: cnd/src/swap_protocols/rfc003/create_swap.rs [hunk 2]
```diff
@@ -317,45 +321,49 @@ where
...
-    <AL as Ledger>::Transaction,
-    <BL as Ledger>::HtlcLocation,
-    <BL as Ledger>::Transaction,
-    AA,
-    BA,
+pub type SwapEventOnLedger<AL, BL, AA, BA, AH, BH> =
+    SwapEvent<AA, BA, AH, BH, <AL as Ledger>::Transaction, <BL as Ledger>::Transaction>;
+
+// Needed to satisfy clippy, this goes away once Ledger::Transaction is gone.
+pub type ReplacementSwapEventOnLedger<A> = SwapEvent<
```
Comment (thomaseizinger) [inline cnd/src/swap_protocols/rfc003/create_swap.rs]: :joy:

**Pull request merged**